subprojects/core/src/main/java/org/gradle/caching/local/internal/DirectoryBuildCacheServiceFactory.java

```diff
@@ -58,26 +56,21 @@ public class DirectoryBuildCacheServiceFactory implements BuildCacheServiceFacto
     public DirectoryBuildCacheServiceFactory(
         UnscopedCacheBuilderFactory unscopedCacheBuilderFactory,
         GlobalScopedCacheBuilderFactory cacheBuilderFactory,
-        PathToFileResolver resolver,
         FileAccessTimeJournal fileAccessTimeJournal,
         CacheConfigurationsInternal cacheConfigurations,
         CacheCleanupStrategyFactory cacheCleanupStrategyFactory
     ) {
         this.unscopedCacheBuilderFactory = unscopedCacheBuilderFactory;
         this.cacheBuilderFactory = cacheBuilderFactory;
-        this.resolver = resolver;
         this.fileAccessTimeJournal = fileAccessTimeJournal;
         this.cacheConfigurations = cacheConfigurations;
         this.cacheCleanupStrategyFactory = cacheCleanupStrategyFactory;
     }
 
     @Override
     public BuildCacheService createBuildCacheService(DirectoryBuildCache buildCacheConfig, Describer describer) {
-        Object cacheDirectory = buildCacheConfig.getDirectory();
-        File target;
-        if (cacheDirectory != null) {
-            target = resolver.resolve(cacheDirectory);
-        } else {
```

Comment:
🤔 I wonder if not resolving here is a breaking change. Most of our tests are using absolute file URLs to set up the cache location, so they won't show that breakage. :(